processor/src/main/java/org/mapstruct/ap/internal/model/common/TypeFactory.java

```diff
@@ -238,6 +239,14 @@ else if ( mirror.getKind() == TypeKind.ARRAY ) {
                 qualifiedName = componentTypeElement.getQualifiedName().toString() + arraySuffix;
                 isImported = isImported( name, qualifiedName );
             }
+            else if (componentTypeMirror.getKind().isPrimitive()) {
+                // When the component type is primitive and is annotated with ElementType.TYPE_USE then
+                // the typeMirror#toString returns (@CustomAnnotation :: byte) for the javac compiler
+                name = componentTypeMirror.getKind().name().toLowerCase( Locale.ROOT ) + builder.toString();
```

Comment:
consider adding a map to `org.mapstruct.ap.internal.util.NativeTypes`. Mapping `TypeKind` to native type instead of doing string conversion.